tests/remote_point_evaluation/distributed_compute_intersection_locations_02.cc

```diff
@@ -0,0 +1,147 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2021 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+// FIle 3
+// check intersections are empty for J<1e-12 on all processors
+// File 4
+// ensure Convsersion to RPE is Correct
+
+#include "../tests.h"
+
+#include "deal.II/grid_tools.h"
+
+using namespace dealii;
+
+template <int dim>
+std::vector<std::vector<BoundingBox<dim>>>
+get_global_bboxes(const parallel::distributed::Triangulation<dim> &tria,
+                  const Mapping<dim> &                             mapping,
+                  const unsigned int rtree_level = 0)
+{
+  std::vector<dealii::BoundingBox<dim>> local_boxes;
+  for (const auto &cell : tria.active_cell_iterators())
+    if (cell->is_locally_owned())
+      local_boxes.emplace_back(mapping.get_bounding_box(cell));
+
+  // create r-tree of bounding boxes
+  const auto local_tree = pack_rtree(local_boxes);
+
+  // compress r-tree to a minimal set of bounding boxes
+  std::vector<std::vector<BoundingBox<dim>>> global_bboxes(1);
+  global_bboxes[0] = extract_rtree_level(local_tree, rtree_level);
+
+  return global_bboxes;
+}
+
+
+
+template <int structdim, int dim>
+void
+do_test(const use_marked_vertices = false)
+{
+  // create triangulation
+  parallel::distributed::Triangulation<dim> tria(MPI_COMM_WORLD);
+  GridGenerator::hyper_cube(tria);
+  tria.refine_global(2);
+
+  // create cache
+  MappingQ1<dim>              mapping;
+  const GridTools::Cache<dim> cache(tria, mapping);
+
+  // create intersection requests
+  std::vector<std::vector<Point<dim>>> intersection_requests;
+  // create intersection request spanning from x=0 to x=1
+  if (Utilities::MPI::this_mpi_process(MPI_COMM_WORLD) == 0)
+    {
+      Triangulation<dim> tria;
+      GridGenerator::hyper_cube(tria, 0.5, 0.5);
+      const auto cell = tria.begin_active();
+
+      Triangulation<dim> tria_temp;
+      GridGenerator::hyper_cube(tria_temp, 0.5, 1.0);
+      Triangulation<dim> simplex_tria;
+      GridGenerator::convert_hypercube_to_simplex_mesh(tria_temp, simplex_tria);
+      const auto simplex_cell = tria.begin_active();
+
+      if (structdim == dim) // cell
+        {
+          intersection_requests.emplace_back(
+            CGALWrappers::get_vertices_in_cgal_order(cell, mapping));
```

Comment:
Would it be possible to swap the vertices inside of the new function? Using deal.II order by the user would be better.

Reply:
Like this [20e74ce](https://github.com/dealii/dealii/pull/15165/commits/20e74ceb901a4a43b77dbdeb35abc068d783bd30)?

Reply:
Yes :+1: